lib/vagrant-omnibus/action/install_chef.rb

```diff
@@ -214,7 +214,7 @@ def fetch_or_create_install_script(env)
                   set version=%1
                   set dest=%~dp0chef-client-%version%-1.windows.msi
                   echo Downloading Chef %version% for Windows...
-                  powershell -command "(New-Object System.Net.WebClient).DownloadFile('#{url}?v=%version%', '%dest%')"
+                  powershell -command "if(test-path '#{url}/chef-client*%version%*.msi'){[array]$tempstore = dir #{url} | ?{$_.name -like 'chef-client*%version%*.msi'};copy $tempstore[0].fullname %dest%}else{(New-Object System.Net.WebClient).DownloadFile('#{url}?v=%version%', '%dest%')}"
```

Comment:
I would do something like 
```
[uri]$uri='#{url}';switch -regex ($uri.scheme){('http(s*)'{$source='#{url}v=%version%'}'default{[$source=(resolve-path (join-path '#{url}' 'chef-client*%version%*.msi')).path)} };(new-object Net.Webclient).DownloadFile(([uri]$source).absoluteuri, '%dest%')}
```
for the PowerShell command.  Casting it to a URI allows you to determine if it is http(s) or a file path (or no scheme if it is a partial path), then you can determine how to append the version details.  Using join-path and resolve-path can assemble the file or unc path appropriately and the casting the result of that to a uri allows either the http or file scheme to be used with webclient (no need to switch back and forth between copy or webclient download.